landoapi/api/try_push.py

```diff
@@ -98,7 +89,39 @@ def parse_revisions_from_request(
     """Convert a set of base64 encoded patches to `Revision` objects."""
     patches_io = (io.StringIO(decode_json_patch_to_text(patch)) for patch in patches)
 
-    patch_helpers = (PATCH_HELPER_MAPPING[patch_format](patch) for patch in patches_io)
+    try:
+        patch_helpers = [
+            PATCH_HELPER_MAPPING[patch_format](patch) for patch in patches_io
+        ]
+    except ValueError as exc:
+        raise ProblemException(
+            400,
+            "Improper patch format.",
+            f"Patch does not match expected format `{patch_format.value}`: {str(exc)}",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
+
+    try:
+        errors = PushAssessor(patch_helpers=patch_helpers, repo=repo).run_push_checks()
```

Comment:
I think it makes more sense to initialize the `PushAssessor` object with the relevant checks, then call `run_push_checks`. This would also remove the need for the `check.relevant` property, unless there are more specific checks of relevancy in the future (though the docstring you have implies that repo-relevancy is the only thing being checked.)

```python
assessor = PushAssessor(patch_helpers=patch_helpers, repo=repo, checks=[BugReferencesCheck])
errors = assessor.run_push_checks()
```